pkg/lib/indexer/indexer.go

```diff
@@ -283,3 +292,182 @@ func write(dockerfileText, outDockerfile string, logger *logrus.Entry) error {
 
 	return nil
 }
+
+// ExportFromIndexRequest defines the parameters to send to the ExportFromIndex API
+type ExportFromIndexRequest struct {
+	Index         string
+	Operator      string
+	DownloadPath  string
+	ContainerTool string
+}
+
+// ExportFromIndex is an aggregate API used to specify operators from
+// an index image
+func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
+	databaseFile := defaultDatabaseFile
+
+	// set a temp directory
+	workingDir, err := ioutil.TempDir("./", "index_tmp")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(workingDir)
+
+	// Pull the fromIndex
+	if request.Index != "" {
+		i.Logger.Infof("Pulling previous image %s to get metadata", request.Index)
+
+		// Get the old index image's dbLocationLabel to find this path
+		labels, err := i.LabelReader.GetLabelsFromImage(request.Index)
+		if err != nil {
+			return err
+		}
+		if dbLocation, ok := labels[containertools.DbLocationLabel]; ok {
+			i.Logger.Infof("Previous db location %s", dbLocation)
+
+			// extract the database to the file
+			err = i.ImageReader.GetImageData(request.Index, workingDir)
+			if err != nil {
+				return err
+			}
+
+			databaseFile = path.Join(workingDir, dbLocation)
+		}
+	} else {
+		databaseFile = path.Join(workingDir, databaseFile)
+	}
+
+	bundles, err := getBundlesToExport(databaseFile, request.Operator)
+	if err != nil {
+		return err
+	}
+	i.Logger.Infof("Preparing to pull bundles %+q", bundles)
+
+	// Creating downloadPath dir
+	if _, err := os.Stat(request.DownloadPath); err != nil {
+		if os.IsNotExist(err) {
+			err = os.Mkdir(request.DownloadPath, 0777)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	var errs []error
+
+	db, err := sql.Open("sqlite3", databaseFile)
+	if err != nil {
+		return err
+	}
+
+	dbQuerier := sqlite.NewSQLLiteQuerierFromDb(db)
+	if err != nil {
+		return err
+	}
+
+	for _, bundleImage := range bundles {
+		// try to name the folder
+		folderName, err := dbQuerier.GetBundleVersion(context.TODO(), bundleImage)
+		if err != nil {
+			return err
+		}
+		if folderName == "" {
+			// operator-registry does not care about the folder name
+			folderName = bundleImage
+		}
+		exporter := bundle.NewSQLExporterForBundle(bundleImage, filepath.Join(request.DownloadPath, folderName), request.ContainerTool)
+		if err := exporter.Export(); err != nil {
+			err = fmt.Errorf("error exporting bundle from image: %s", err)
+			errs = append(errs, err)
+		}
+	}
+	err = db.Close()
+	if err != nil {
+		errs = append(errs, err)
+		return utilerrors.NewAggregate(errs)
+	}
```

Comment:
Does this need to handle bundles (non-image) as well, or are we sure packages will be migrated to all-image?

Reply:
IMO it doesn't make sense to support non-image bundles since these cannot create an index (as the opm cli is currently defined). In the future, if we support inserting non-image bundles into an index then we will probably need to add non-image bundle support for exporting.

Reply:
It depends on how they will be bootstrapped, correct?

I can take a database built with `initializer` and then pass it to `opm` to add another entry to it.

Reply:
No doubt that this is possible but this is not how we want to or will bootstrap them